torch/testing/_internal/common_utils.py

```diff
@@ -980,17 +981,17 @@ def _compareScalars(self, a, b, *, rtol=None, atol=None, equal_nan=True):
                 rtol, atol = 0, 0
         atol = max(atol, self.precision)
 
-        return _compare_scalars_internal(a, b, rtol=rtol, atol=atol, equal_nan=equal_nan)
+        return _compare_scalars_internal(a, b, rtol=cast(float, rtol), atol=cast(float, atol), equal_nan=equal_nan)
 
-    def assertEqualIgnoreType(self, *args, **kwargs):
+    def assertEqualIgnoreType(self, *args, **kwargs) -> None:
         # If you are seeing this function used, that means test is written wrongly
         # and deserves detailed investigation
         return self.assertEqual(*args, exact_dtype=False, **kwargs)
 
     # Compares x and y
     # TODO: default exact_device to True
-    def assertEqual(self, x, y, *, atol=None, rtol=None, equal_nan=True,
-                    exact_dtype=True, exact_device=False, msg=None):
```

Comment:
This implementation of `assertEqual` overrides `unittest.TestCase.assertEqual(self, x, y, msg=None)`, which is a violation of LSP

Reply:
Updated PR to make it compatible.